streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java

```diff
@@ -762,6 +762,9 @@ private KafkaStreams(final InternalTopologyBuilder internalTopologyBuilder,
             storeProviders.add(new StreamThreadStateStoreProvider(threads[i], internalTopologyBuilder));
         }
 
+        ClientMetrics.addNumAliveStreamThreadMetric(streamsMetrics, (metricsConfig, now) ->
+            Math.toIntExact(Arrays.stream(threads).filter(thread -> thread.state().isAlive()).count()));
```

Comment:
req: Please add verifications or unit tests to verify the values measured by this metric in `KafkaStreamsTest`.